lib/Driver/Driver.cpp

```diff
@@ -1057,7 +1057,17 @@ void Driver::buildOutputInfo(const ToolChain &TC, const DerivedArgList &Args,
 
   if (driverKind == DriverKind::Interactive) {
     OI.CompilerMode = OutputInfo::Mode::Immediate;
-    if (Inputs.empty())
+    /* SR-5860: For "-e" we need to stay in Immediate mode, otherwise down in
+     buildActions(...) we construct a REPLJobAction instead of the intended
+     InputJobAction, which later results in us dropping down into the REPL (if
+     I'm correctly understanding how this mechanism works).
+
+     So I **think** we'll need to have this check here because the OutputInfo
+     that's passed in to Driver::buildActions(...) is \const\, thus we can't
+     switch it to Immediate from within there (and demoting it to a variable
+     in order to mutate it within buildActions is wrong in my book).
+     */
```

Comment:
Yep, your logic is correct. Note that we prefer to always use `//` for comments, even multi-line ones.